### PR TITLE
C++ and Python Solver Profiling

### PR DESCRIPTION
**Pull request by ethangreen-dev**

### CODE REVIEW

File: test/profiling/__init__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class PerformanceEntry:
```
Comment (jtcooper10) [inline test/profiling/__init__.py:1]: Classes defined directly in `__init__.py`, consider placing into a separate file

File: test/profiling/python_profiler.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    perf_data = PerformanceData()
+    perf_data.perf_time = stats.total_tt
+    perf_data.call_time = 0
+    
+    worst_func = stats.fcn_list[0]
+    (cc, nc, tt, ct, callers) = stats.stats[worst_func]
+    perf_data.worst_entry = (
+        worst_func, 
+        PerformanceEntry(round(tt, 3), round((tt / stats.total_tt) * 100))
```
Comment (jtcooper10) [inline test/profiling/python_profiler.py:40]: Consider leaving the contents of the `PerformanceEntry` unrounded, and instead round on each use.

File: test/profiling/__init__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        self.percent = float(percent)
+
+class PerformanceData:
+    perf_time: float = -1.0
+    call_time: float = -1.0
+    call_list: "dict[str, PerformanceEntry]" = {}
+    worst_entry: "tuple[str, PerformanceEntry]" = ("unknown", PerformanceEntry())
+
+    def __str__(self):
+        string = f"Total time: {round(self.perf_time, 6)}ms ({self.call_time}ms sampled)\n"
```
Comment (mdip226) [inline test/profiling/__init__.py:16]: Maybe clarify what these values represent
Like, "Program Execution Time" and "Solver Execution Time" or something

**Review by mdip226 — APPROVED** [review]

**Review by jtcooper10 — APPROVED** [review]

**Review by briandrawert — APPROVED** [review]

**Review by BryanRumsey — APPROVED** [review]
Looks good

**Pull request merged**